### ISSUE TITLE
[A11y] General - Focus Visibility

### ISSUE THREAD

**Issue by jechols** [OregonDigital/OD2#1921]
### Operating System

Windows

### Browser

Firefox

### Affected page

Homepage

### Admin user?

Yes

### Issue Description

Using Edge for most testing, I only just discovered that in Firefox some controls actually seem to be hiding focus indicators.  Edge seems to force a focus indicator around elements even when CSS would otherwise hide such an indicator, which is lovely for Edge users but awful for audits.

- The logo at the top has no focus indicator
- Whenever there's a "current" button (grid vs. list views), with the white on orange, it has no focus indicator
- Explore Collections has a focus indicator on the top buttons ("ALL", "OREGON STATE UNIVERSITY", etc.), but it fails to change anything other than color, and only barely.  I don't recall which WCAG area that fails, but it fails one of them....
- The fake dropdowns (e.g., "results per page") only change their border color and only slightly
- The "MY ACCOUNT" button might pass, I'm not sure, but when it's "pressed" it looks exactly the same as when it has focus, which is a separate kind of failure, so its focus should be indicated differently.

If this needs to be multiple tickets I can split that up, I'm just not sure which pieces will be easier to handle as a single change....

### Suggested Fix

Make sure keyboard focus is always very clear.  If there is a hover style, it's usually good to apply that, although in many of the above cases the hover style doesn't seem useful, either.

For something like the "grid" or "masonry" or "list" buttons, a 2px or so black border around the element would be fine.  For the buttons which already have borders around them, something a bit more drastic is required, like inverting the colors or else adding an inversion of the border, but with enough of the original border that it is noticeable and doesn't just get lost.  Edge does something like that and it seems okay, though I honestly have no idea how that's accomplished.

### Can anybody QA or does this need an expert?

Anybody

### QA Steps

Use Firefox and tab to every single control across all pages.  Yeah.

**Comment by azle** [comment]
Updated 12/30/2024 In Progress

FIREFOX
- The logo at the top has no focus indicator FIXED
- Whenever there's a "current" button (grid vs. list views), with the white on orange, it has no focus indicator FIXED
- Explore Collections has a focus indicator on the top buttons ("ALL", "OREGON STATE UNIVERSITY", etc.), but it fails to change anything other than color, and only barely. I don't recall which WCAG area that fails, but it fails one of them....FIXED (HAS HIGHLIGHT AND READS ARIA LABEL)
- The fake dropdowns (e.g., "results per page") only change their border color and only slightly FIXED (HAS HIGHLIGHT AND READS ARIA LABEL)
- The "MY ACCOUNT" button might pass, I'm not sure, but when it's "pressed" it looks exactly the same as when it has focus, which is a separate kind of failure, so its focus should be indicated differently. FIXED (HAS HIGHLIGHT AND DROPDOWM MENU)
- On the "CREATE A NEW COLLECTION" modal, the visible focus indicator around "SAVE" is barely visible on Firefox. The border changes almost imperceptibly, and the background color moves from a very dark blue to black. FIXED (HAS HIGHLIGHT AND BUTTON CHANGES FROM BLUE TO BLACK)
- On the asset view page, the "FILE MANAGER" button has very little to indicate it's got focus, even with Edge trying its hardest to force a focus rectangle around it. @jsimic this must be old I dont see the File Manager when I am logged in and looking at a specific item in a collection. Can you verify I am looking at the right place. Here is the link I am using logged in: https://oregondigital.org/concern/images/zs25x9840?locale=en
